todo/src/reducer/index.js

```diff
@@ -0,0 +1,52 @@
+import {ADD_ITEM, MARK_ITEM, UNMARK_ITEM} from '../actions/App.js';
+
+const initialState = {
+    listItems: [{
+        text:"placeholder item",
+        completed: false,
+        id: 0
+    }]
+};
+
+const reducer = (state = initialState, action) =>{
+    //console.log('made it to reducer')
+
+    switch (action.type){
+        case ADD_ITEM:
+            return{
+                ...state,
+                listItems: [
+                    ...state.listItems, 
+                    action.payload
+                ]
+            };
+        case MARK_ITEM:
+            //console.log("made it to mark item");
+            const newArray = state.listItems.map(item =>{
```

Comment:
this is awesome and totally works but you could shrink it down to one case if you wanted. something else to note is when you set item.completed to true you're directly mutating that state array. map will make a shallow copy (of the array itself) but the objects within the array are still references. to fix that you could use a second spread operator. this would all look something like:
```return {...state, listItems: state.listItems.map(todo => todo.id === action.payload ? {...todo, completed: !todo.completed} : todo) }```